android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -282,27 +335,89 @@ public void onError(String info) {
+	 */
+	@Deprecated
+	public void startRemoteDisplayStream(Context context, Class<? extends SdlRemoteDisplay> remoteDisplayClass, VideoStreamingParameters parameters, final boolean encrypted){
+        configureGlobalParameters(context, remoteDisplayClass, isEncrypted, null);
+		boolean isCapabilitySupported = internalInterface.getSystemCapabilityManager() != null && internalInterface.getSystemCapabilityManager().isCapabilitySupported(SystemCapabilityType.VIDEO_STREAMING);
+		if(majorProtocolVersion >= 5 && !isCapabilitySupported){
+			DebugTool.logError(TAG, "Video streaming not supported on this module");
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+			return;
+		}
+		if (!HMILevel.HMI_NONE.equals(currentOnHMIStatus.getHmiLevel()) && VideoStreamManager.this.parameters == null) {
+			getVideoStreamingParams();
+		}
+		checkState();
+		processCapabilitiesWithPendingStart(encrypted, parameters);
+	}
+
+	private void configureGlobalParameters(Context context, Class<? extends SdlRemoteDisplay> remoteDisplayClass, boolean encrypted, List<VideoStreamingRange> streamingRange) {
+		this.context = new WeakReference<>(context);
+		this.remoteDisplayClass = remoteDisplayClass;
+		this.isEncrypted = encrypted;
+		this.majorProtocolVersion = internalInterface.getProtocolVersion().getMajor();
+		if (streamingRange != null) {
+			this.streamingRange = streamingRange;
+		}
+	}
+
+	private void processCapabilitiesWithPendingStart(boolean encrypted, VideoStreamingParameters parameters){
         if (parameters == null) {
             if (majorProtocolVersion >= 5) {
                 if (internalInterface.getSystemCapabilityManager() != null) {
                     internalInterface.getSystemCapabilityManager().getCapability(SystemCapabilityType.VIDEO_STREAMING, new OnSystemCapabilityListener() {
                         @Override
                         public void onCapabilityRetrieved(Object capability) {
                             VideoStreamingParameters params = new VideoStreamingParameters();
-                            params.update((VideoStreamingCapability) capability, vehicleMake);    //Streaming parameters are ready time to stream
-                            startStreaming(params, encrypted);
+                            VideoStreamingCapability castedCapability = ((VideoStreamingCapability)capability);
+						VideoStreamManager.this.originalCapability = castedCapability;
+
+						// Mocks data here
+						// castedCapability.setAdditionalVideoStreamingCapabilities(getMockedAdditionalCapabilities());
+						params.update(castedCapability, vehicleMake);    //Streaming parameters are ready time to stream
+                            VideoStreamManager.this.parameters = params;
+
+						if (streamingRange != null) {
+							// filtering
+							castedCapability.setAdditionalVideoStreamingCapabilities(
+									getSupportedCapabilities(
+											streamingRange,
```

Comment:
@kostyaBoss on line 410 you can change this from streamingRange to listOfStreamingRanges.